internal/operatornetworkpolicies/operatornetworkpolicies.go

```diff
@@ -0,0 +1,108 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package operatornetworkpolicies
+
+import (
+	"context"
+
+	corev1 "k8s.io/api/core/v1"
+	networkingv1 "k8s.io/api/networking/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/util/intstr"
+	"k8s.io/client-go/kubernetes"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+	"sigs.k8s.io/controller-runtime/pkg/manager"
+)
+
+const (
+	operatorName = "opentelemetry-operator-controller-manager"
+)
+
+type networkPolicies struct {
+	clientset         kubernetes.Interface
+	operatorNamespace string
+	scheme            *runtime.Scheme
+}
+
+var _ manager.Runnable = (*networkPolicies)(nil)
+var _ manager.LeaderElectionRunnable = (*networkPolicies)(nil)
+
+func NewOperatorNetworkPolicies(operatorNamespace string, clientset kubernetes.Interface, scheme *runtime.Scheme) manager.Runnable {
+	return &networkPolicies{
+		clientset:         clientset,
+		operatorNamespace: operatorNamespace,
+		scheme:            scheme,
+	}
+}
+
+func (n *networkPolicies) Start(ctx context.Context) error {
+	tcp := corev1.ProtocolTCP
+	webhookPort := intstr.FromInt32(9443)
+	metricsPort := intstr.FromInt32(8443)
```

Comment:
Can this lead to issues when users configure different ports?
The helm chart default seems to look like this:
```
    Args:
      --metrics-addr=0.0.0.0:8080
      --enable-leader-election
      --health-probe-addr=:8081
      --webhook-port=9443
      --collector-image=otel/opentelemetry-collector-k8s:0.129.1
```

Reply:
I have added dynamic port setting from the supplied operator config